docs/en/sql-reference/functions/other-functions.md

```diff
@@ -2984,6 +2984,66 @@ Result:
 └─────────┘
 ```
 
+## partitionId
```

Comment:
I played around with this function (didn't know it existed) but I could not find a real use case for it.

E.g. in https://fiddle.clickhouse.com/c70544e4-ca6a-4d8f-8162-42ef3d201da2 one could get the same result by SELECTing `_partition_id` or the partition expression directly. From a user perspective, `_partition_id` is certainly easier to use.

My proposal is to remove this function (it has never been documented) and adjust integration/test_lost_part/test.py and queries/0_stateless/01748_partition_id_pruning.sql (the only two tests using `partitionId`) to use `_partition_id`.

Reply:
> but I could not find a real use case for it.

It's exist.
Algorithm for generation of partitionId is "kind of obscured" in CHI, so if you want to understand, in which partitionId this row will land, you forced to "research and reimplement such algo" by yourself

And this function is helpful, when you build service, which need to manipulate with partitions. for example, we done that for clickhouse-backup.
Instead you can just run simple "SELECT partitionId(...)" and be sure that you get correct result.

https://github.com/ClickHouse/ClickHouse/issues/46584#issuecomment-1530932400